cvat/apps/engine/ddln/tasks/vls_lines/models.py

```diff
@@ -0,0 +1,139 @@
+        else:
+            self._check_parallel_lon_order(reporter)
+        self._check_lat_order(reporter)
+
+    def apply_visibility(
+        self,
+        left_visible,
+        right_visible,
+        center_visible,
+        start_visible,
+        end_visible,
+        designator_visible
+    ):
+        if not left_visible:
+            self.left_line = None
+        if not right_visible:
+            self.right_line = None
+        if not center_visible:
+            self.center_line = None
+        if not start_visible:
+            self.start_line = None
+        if not end_visible:
+            self.end_line = None
+        if not designator_visible:
+            self.designator_line = None
+
+    def _check_lon_order(self, reporter):
+        original = [self.left_line, self.center_line, self.right_line]
+        max_angle = 0
+        edges = None
+        for a, b in combinations(original, 2):
+            angle = get_lines_angle(a, b)
+            if angle > max_angle:
+                max_angle = angle
+                edges = [a, b]
+        new_center = next(filter(lambda line: line not in edges, original))
+        new_left, new_right = edges
+        if get_counterclockwise_angle(new_right.get_angle(), new_left.get_angle()) >= math.pi:
+            new_left, new_right = new_right, new_left
+        final = [new_left, new_center, new_right]
+        if final != original:
+            reporter.report_lon_disorder()
+
+    def _check_parallel_lon_order(self, reporter):
+        large_step = abs(self.left_line.c - self.right_line.c)
+        small_step_a = abs(self.center_line.c - self.left_line.c)
+        small_step_b = abs(self.center_line.c - self.right_line.c)
+        if small_step_a >= large_step or small_step_b >= large_step:
+            reporter.report_lon_disorder()
+
+    def _check_lat_order(self, reporter):
+        original = [self.end_line, self.designator_line, self.start_line]
+        if any(line is None for line in original):
+            return
+        distant_point = self.lon_vanishing_point
+        if not distant_point:
+            # if longitudinal lines are parallel, just assume they are directed away from the camera
+            distant_point = Line.by_point_and_angle(Point(0, 0), 0).intersect(self.center_line)
+        if not distant_point:
+            distant_point = Point(0, -10000)
+        final = sorted(original, key=lambda line: distant_point.distance_to(line))
+        if final != original:
+            reporter.report_lat_disorder()
+
+    def _calculate_vanishing_point(self, lines, reporter, is_lon):
+        combs = ((a.intersect(b), get_lines_angle(a, b)) for a, b in combinations(lines, 2))
+        points, angles = zip(*combs)
+        points = [p for p in points if p]
+        if not points:
+            return None
+        max_angle = max(angles)
+        if max_angle <= ANGLE_THRESHOLD:
+            return None
+
+        average_point = get_average_point(points)
```

Comment:
Taking average is only ok because we know that the Fan tool should enforce them being essentially the same.
I'd add a check that they are close to each other just to catch bugs in the Fan tool.